File: 0234: palindromeLinkedList.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        
        stack<int> check;
        auto current = head;
        while(current){

            check.push(current->val);
            current = current->next;
        }

        current = head;
        while(!check.empty()){

            if(check.top() != current->val){

                return false;
            }
            //cout << check.top() << ' ' << current->val <<endl;
            check.pop();
            current = current->next;
        }

        return true;
    }
};

File: 0234: palindromeLinkedList.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        nums = []
        current = head
        while current:
            nums.append(current.val)
            current = current.next
        i = 0
        j = len(nums)-1
        while i < j:
            if nums[i] != nums[j]:
                return False
            i+=1
            j-=1
        return True